src/odemis/gui/model/__init__.py

```diff
@@ -626,6 +626,39 @@ def __init__(self, main):
         self.spec_bck_cal.subscribe(self._on_spec_bck_cal)
         self.spec_cal.subscribe(self._on_spec_cal)
 
+        self.zpos = model.FloatContinuous(0, range=(0, 0), unit="m")
+        self.zpos.clip_on_range = True
+        self._updateZParams()
+        self.streams.subscribe(self._on_stream_change, init=True)
+
+    def _updateZParams(self):
+        if not any(model.hasVA(s, "zindex") for s in self.streams.value):
+            return
+
+        # Calculate the new range of z pos
+        lo, hi = 0, 0
```

Comment:
This seems quite broken. Does it work if the lowest is > 0? Does it work if the highest is  < 0? I'd recommend to just use `min()` on all the range mins (or maxs) at the same time.

Also, I don't think we should handle negative zstep. => Just detect that and log a warning in such cases.

Also, `MD_POS` indicates the _center_, not the start.